state-sync/diem-data-client/src/diemnet/mod.rs

```diff
@@ -460,10 +471,22 @@ impl DataSummaryPoller {
                             .message("Unable to select next peer")
                             .error(&error))
                     );
+                    increment_counter(
+                        &metrics::PEER_SELECTION_ERRORS,
+                        StorageServiceRequest::GetStorageServerSummary
+                            .get_label()
+                            .into(),
+                    );
                     continue;
                 }
             };
 
+            let _timer = start_timer(
```

Comment:
`_timer` will only be dropped at the end of this loop (when it falls out of scope), meaning it will also time `update_summary` and `update_global_summary`. To avoid this we should drop it manually after we get a response e.g., add `drop(timer)` to line 498, after we get the result?

Reply:
Added `drop(timer)` right after receiving the response (and before `update_summary` and `update_global_summary`)

Reply:
I thought that `_` were also not dropped until the end of the scope (I may be wrong).  But, the whole point of it all is that `_x` or `_` doesn't bind the variable to anything and can't be used.

It's a very common pattern for anything that's needed for the scope, e.g. a lock or a timer, to do this and then for the entirety of a scope `{ ... }` be held and then dropped. More info about dropping and scope here: https://doc.rust-lang.org/book/ch15-03-drop.html

Reply:
Changed `_timer` to `timer`.

Thanks @gregnazario and @JoshLind for giving me more pointers to rust naming convention and dropping!

n00b question: I tried to change `_timer = start_timer(...)` to `_ = start_timer(...)` in another place where `_timer` is automatically dropped at the end of the scope. Though the compiler did not complain, this change would result in no latency being calculated since no actual timer object is bound to be able to stop? (and hence that's why we'd need to use `_timer` instead of `_`?)

Reply:
The compiler doesn't have an idea of what you're doing.  It's often legitimate to say `let _ = function()` just to use a "must use" return value (and I guess drop it).

So, that looks right then, you must use a name to keep it around for the scope if it's timing nothing.

Reply:
Yeah got it. Thanks @gregnazario! 😃